--- beetle-saturn-libretro-f7515829166660a61c6692160bb2b2d2633df9fc-orig/libretro-common/rthreads/rthreads.c
+++ beetle-saturn-libretro-f7515829166660a61c6692160bb2b2d2633df9fc/libretro-common/rthreads/rthreads.c
@@ -865,7 +865,7 @@
    int tickms = ps2_clock();
    now.tv_sec = tickms/1000;
    now.tv_nsec = tickms * 1000;
-#elif defined(__mips__) || defined(VITA) || defined(_3DS)
+#elif defined(VITA) || defined(_3DS)
    struct timeval tm;
 
    gettimeofday(&tm, NULL);
