--- RetroArch-1.9.13/libretro-common/rthreads/rthreads.c.orig
+++ RetroArch-1.9.13/libretro-common/rthreads/rthreads.c
@@ -872,7 +872,7 @@
    int tickms = ps2_clock();
    now.tv_sec = tickms/1000;
    now.tv_nsec = tickms * 1000;
-#elif !defined(DINGUX_BETA) && (defined(__mips__) || defined(VITA) || defined(_3DS))
+#elif !defined(DINGUX_BETA) && (defined(VITA) || defined(_3DS))
    struct timeval tm;
 
    gettimeofday(&tm, NULL);
