--- melonDS-657c729e7473e9b2c09ddf2eef9dcf4ba5b247ec/src/libretro/libretro-common/rthreads/rthreads.c.orig
+++ melonDS-657c729e7473e9b2c09ddf2eef9dcf4ba5b247ec/src/libretro/libretro-common/rthreads/rthreads.c
@@ -873,7 +873,7 @@
    int tickms = clock();
    now.tv_sec = tickms/1000;
    now.tv_nsec = tickms * 1000;
-#elif defined(__mips__) || defined(VITA) || defined(_3DS)
+#elif defined(VITA) || defined(_3DS)
    struct timeval tm;
 
    gettimeofday(&tm, NULL);
